src/Oscoin/Time/Chrono.hs

```diff
@@ -0,0 +1,42 @@
+{-- | Little helpers geared towards making obvious in the types
+      how blocks are ordered.
+--}
+
+module Oscoin.Time.Chrono
+    ( NewestFirst(..)
+    , OldestFirst(..)
+
+    , reverse
+    ) where
+
+import qualified Data.List as List
+import qualified Data.List.NonEmpty as NE
+import           Oscoin.Prelude hiding (reverse)
+
+
+class ChronogicallyOrdered a where
+    type ReverseOf a :: *
+    reverse :: a -> ReverseOf a
```

Comment:
It looks to me that `a` must be either `NewestFirst f` or `OldestFirst f`. Who about making this class less general
~~~haskell
class ChronologicallyOrdered f where
  toOldestFirst :: NewestFirst f -> OldestFirst f
  toNewestFirst :: OldestFirst f -> NewestFirst f

Reply:
Interesting suggestion -- the only pain in the next is that now `toOldestFirst` and `toNewestFirst` clash with the "deconstructors", so I will have to rename them `getOldestFirst` and `getNewestFirst` and amend the code throughout, but it can be done 😀